fedimint-client/src/lib.rs

```diff
@@ -860,6 +903,17 @@ struct ClientInner {
     root_secret: DerivableSecret,
     operation_log: OperationLog,
     secp_ctx: Secp256k1<secp256k1_zkp::All>,
+    /// Task group used by this client and shut down when the client is dropped.
+    /// If a subgroup is used, dropping the parent task group stops the client's
+    /// background tasks.
+    tg: Mutex<Option<TaskGroup>>,
+    /// Number of [`Client`] instances using this `ClientInner`.
+    ///
+    /// The `ClientInner` struct is both used for the client itself as well as
```

Comment:
Hmmmm... OK

Reply:
I also tried to make all the context uses of `ClientInner` weak, but that was very inelegant (how to propagate up the error case when upgrading?) and lead to weird errors that I gave up debugging since I had wasted too much time on this already.